db/db_impl.h

```diff
@@ -992,9 +992,14 @@ class DBImpl : public DB {
       writer = nullptr;
       return w;
     }
-    void ClearWriter() {
+    Status ClearWriter() {
+      Status s = writer->WriteBuffer();
+      if (s.ok()) {
+        s = writer->file()->Sync(true);
```

Comment:
This is new. Can you explain more why we need to add this?

Reply:
This will do an fsync on the file. WriteBuffer will eventually call fdatasync which doesn't update metadata such as atime. Although now that I think about it, we should just call file()->Sync() instead of WriteBuffer, which will do all the work.

Reply:
Normally fsync is done in the background thread as the first thing of memtable flush. This will introduce high latency for normal write.
This fsync should only be called in DB close case.

Reply:
I think the WriteBuffer() status code that we swallowed is this one: https://github.com/facebook/rocksdb/blob/0f0d2ab95a2107256498e7be2bbd3d4e8c0c7c32/db/db_impl_write.cc#L1314